### PR TITLE
Adding Helix testing infrastructure

### PR DESCRIPTION
**Pull request by joperezr**
cc: @safern @krwq @ulisesh @joshfree 

Adding Helix infrastructure to IoT. With these changes, we will start running helix hardware tests as part of CI on raspberry pis. There are three major things that need to happen first before we can merge this PR:

 - [x] We need a new version of helix sdk since I recently merged a fix required for our tests.
 - [ ] ~~I need to add here the yml changes so that the test build is triggered as part of CI.~~
 - [ ] ~~We need to make sure that the lab has the hardware setup correctly so that we can start submitting jobs to the queue.~~
_Merging as soon as CI is green. We will do steps 2 and 3 on a different PR._

I'll mark this PR as No Merge for now until the three above are completed, at which point I'll remove the label and we can go ahead and merge.

### CODE REVIEW

File: eng/sendToHelix.proj [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  </ItemGroup>
+
+  <PropertyGroup Condition=" '$(HelixAccessToken)' == '' ">
+    <IsExternal>true</IsExternal>
+    <Creator>$(BUILD_SOURCEVERSIONAUTHOR)</Creator>
+    <Creator Condition=" '$(Creator)' == ''">anon</Creator>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(HelixAccessToken)' == '' ">
+    <HelixTargetQueue Include="Raspbian.9.Arm32.IoT.open"/>
```
Comment (joperezr) [inline eng/sendToHelix.proj]: @ulisesh Is this queue ready as well? We will need it for CI jobs.

**Comment by joperezr** [comment]
> or you can merge and send random PR which proves nothing is broken

Once I add the yml changes (second bullet point on the description) this PR will actually be running the tests on Helix already, so I won't be able to merge with a green CI unless nothing is broken 😄

File: eng/sendToHelix.proj [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    those tests too. -->
+    <XUnitProject Include="..\src\System.Device.Gpio.Tests\System.Device.Gpio.Tests.csproj">
+      <TargetFramework>netcoreapp2.1</TargetFramework>
+      <RuntimeTargetFramework>netcoreapp2.0</RuntimeTargetFramework>
+      <AdditionalProperties>RuntimeIdentifier=linux-arm</AdditionalProperties>
+    </XUnitProject>
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(HelixAccessToken)' != '' ">
+    <HelixTargetQueue Include="Raspbian.9.Arm32.IoT"/>
```
Comment (safern) [inline eng/sendToHelix.proj]: What if you want to do local testing and send to another queue?

File: eng/sendToHelix.proj [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  </ItemGroup>
+
+  <PropertyGroup Condition=" '$(HelixAccessToken)' == '' ">
+    <IsExternal>true</IsExternal>
+    <Creator>$(BUILD_SOURCEVERSIONAUTHOR)</Creator>
+    <Creator Condition=" '$(Creator)' == ''">anon</Creator>
```
Comment (safern) [inline eng/sendToHelix.proj]: This creator name seems odd to me and hard to remember, maybe use dotnet-bot?

File: eng/sendToHelix.proj [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    <HelixTargetQueue Include="Raspbian.9.Arm32.IoT.open"/>
+  </ItemGroup>
+
+  <!-- Our tests require to run as elevated. -->
+  <Target Name="AppendSudoToHelixWorkItems"
+          Inputs="%(HelixWorkItem.Identity)%(HelixWorkItem.Command)"
+          Outputs="unused"
+          AfterTargets="CreateXUnitWorkItems">
+    <PropertyGroup>
+      <HelixPreCommands>$(HelixPreCommands);export _dotnet="%24(which dotnet)"</HelixPreCommands>
```
Comment (safern) [inline eng/sendToHelix.proj]: You will eventually need to check if the targetOS is Windows or not and do it differently on windows. Just as a side note.

File: eng/sendToHelix.proj [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      <_newCommand>sudo -E %24_dotnet $throwAway%(HelixWorkItem.Command)</_newCommand>
```
Comment (safern) [inline eng/sendToHelix.proj]: I don't think you need to escape $ here... 
```suggestion
      <_newCommand>sudo -E $_dotnet $throwAway%(HelixWorkItem.Command)</_newCommand>
```

File: eng/sendToHelix.proj [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        <Command>$(_newCommand)</Command>
+      </HelixWorkItem>
+    </ItemGroup>
+  </Target>
+
+  <!-- Useless stuff to make Arcade SDK happy -->
+  <PropertyGroup>
+    <Language>msbuild</Language>
+  </PropertyGroup>
+  <Target Name="Pack"/>
```
Comment (safern) [inline eng/sendToHelix.proj]: Why do you need to override the Pack target?

File: eng/sendToHelix.proj (hunk 3 above)
Comment (joperezr) [reply eng/sendToHelix.proj]: anon is for anonymous, but I can change it to dotnet-bot as well.

File: eng/sendToHelix.proj (hunk 4 above)
Comment (joperezr) [reply eng/sendToHelix.proj]: We only need to run on sudo on Linux, you can control Gpio on WIndows without being elevated so this target entirely will be conditioned to just run on Linux once we add windows testing.

File: eng/sendToHelix.proj (hunk 6 above)
Comment (joperezr) [reply eng/sendToHelix.proj]: I'm not sure if I need to actually, I was mimicking what arcade UnitTests where doing. I'll try removing and see if it works.

**Pull request merged**